base/cvd/cuttlefish/host/commands/cvd/selector/instance_group_record.cpp

```diff
@@ -16,14 +16,41 @@
 
 #include "host/commands/cvd/selector/instance_group_record.h"
 
+#include <android-base/parseint.h>
+
+#include "common/libs/utils/result.h"
 #include "host/commands/cvd/selector/instance_database_types.h"
 #include "host/commands/cvd/selector/instance_database_utils.h"
 #include "host/commands/cvd/selector/selector_constants.h"
 
 namespace cuttlefish {
 namespace selector {
 
-LocalInstanceGroup::LocalInstanceGroup(const InstanceGroupParam& param)
+namespace {
+
+static constexpr const char kJsonGroupName[] = "Group Name";
+static constexpr const char kJsonHomeDir[] = "Runtime/Home Dir";
+static constexpr const char kJsonHostArtifactPath[] = "Host Tools Dir";
+static constexpr const char kJsonProductOutPath[] = "Product Out Dir";
+static constexpr const char kJsonStartTime[] = "Start Time";
+static constexpr const char kJsonInstances[] = "Instances";
+static constexpr const char kJsonParent[] = "Parent Group";
+
+std::vector<LocalInstance> Filter(
+    const std::vector<LocalInstance>& instances,
+    std::function<bool(const LocalInstance&)> predicate) {
+  std::vector<LocalInstance> ret;
+  for (const auto& instance : instances) {
+    if (predicate(instance)) {
+      ret.push_back(instance);
+    }
+  }
+  return ret;
+}
```

Comment:
This could be

```c++
template<typename T, typename F>
std::vector<T> Filter(const std::vector<T>& in, F pred) {
  std::vector<T> out;
  std::copy_if(in.begin(), in.end(), std::back_inserter(out), pred);
  return out;
}
```
[Compiler explorer](https://godbolt.org/z/Gc1o3dnje)

Reply:
Done